phpBB/phpbb/group/helper.php

```diff
@@ -13,19 +13,73 @@
 
 namespace phpbb\group;
 
+use phpbb\auth\auth;
+use phpbb\cache\service as cache;
+use phpbb\config\config;
+use phpbb\language\language;
+use phpbb\event\dispatcher_interface;
+use phpbb\path_helper;
+use phpbb\user;
+
 class helper
 {
-	/** @var  \phpbb\language\language */
+	/** @var auth */
+	protected $auth;
+
+	/** @var cache */
+	protected $cache;
+
+	/** @var config */
+	protected $config;
+
+	/** @var language */
 	protected $language;
 
+	/** @var dispatcher_interface */
+	protected $dispatcher;
+
+	/** @var path_helper */
+	protected $path_helper;
+
+	/** @var user */
+	protected $user;
+
+	/** @var string phpBB root path */
+	protected $phpbb_root_path;
+
+	/** @var array Return templates for a group name string */
+	protected $name_strings;
+
 	/**
 	 * Constructor
 	 *
-	 * @param \phpbb\language\language $language	Language object
+	 * @param auth					$auth			Authentication object
+	 * @param cache					$cache			Cache service object
+	 * @param config				$config			Configuration object
+	 * @param language				$language		Language object
+	 * @param dispatcher_interface	$dispatcher		Event dispatcher object
+	 * @param path_helper			$path_helper	Path helper object
+	 * @param user					$user			User object
 	 */
-	public function __construct(\phpbb\language\language $language)
+	public function __construct(auth $auth, cache $cache, config $config, language $language, dispatcher_interface $dispatcher, path_helper $path_helper, user $user)
 	{
+		$this->auth = $auth;
+		$this->cache = $cache;
+		$this->config = $config;
 		$this->language = $language;
+		$this->dispatcher = $dispatcher;
+		$this->path_helper = $path_helper;
+		$this->user = $user;
+
+		$this->phpbb_root_path = $path_helper->get_phpbb_root_path();
+
+		$this->name_strings = array(
+			'base_url'				=> append_sid("{$path_helper->get_phpbb_root_path()}memberlist.{$path_helper->get_php_ext()}", 'mode=group&amp;g={GROUP_ID}'),
+			'tpl_noprofile'			=> '<span class="username">{GROUP_NAME}</span>',
```

Comment:
@hanakin I guess this should ideally be refactored out?

Reply:
@marc1706 well ideally none of these would be there, but that means we would have to replace all the uses with the correct format in the templates which will be tedious and would have huge bc concerns most likely. Probably outside the scope of this pr and definitely 3.2.x. Also, I have to double check the CSS for username vs username-colored not sure what exactly the difference between those two are other than for style creators which will lead to customization issues? Might be easier to leave it and just document it clearly so we can easily find and change the html if need be. There are probably a few cases that it may make sense to keep it in the backend like this, but we need to have a clear documented list of the uses. For now, I guess maybe @mrgoldy can add an @html tag comment in the doc block that captures them.

Reply:
What exactly do you want me to put in the DocBlock? If you can let me know, I'll add it.
Do you also want me to add it to the `get_username_string` occurences of these strings ([here](https://github.com/phpbb/phpbb/blob/master/phpBB/includes/functions_content.php#L1485))?

Reply:
Just add @html and a desc of what html is return. For every function that returns html so we can easily find it to factor out later if need be.

Reply:
Well possibly reword it to be “group name and username spans and links” also would consider adding to the doc block as well since the function needs replaced or rewritten essentially.

Reply:
Done